### PR TITLE
Log warnings for numIterations * miniBatchFraction < 1.0

### PR DESCRIPTION
**Pull request by Hydrotoast**
## What changes were proposed in this pull request?

Add a warning log for the case that `numIterations * miniBatchFraction <1.0` during gradient descent. If the product of those two numbers is less than `1.0`, then not all training examples will be used during optimization. To put this concretely, suppose that `numExamples = 100`, `miniBatchFraction = 0.2` and `numIterations = 3`. Then, 3 iterations will occur each sampling approximately 6 examples each. In the best case, each of the 6 examples are unique; hence 18/100 examples are used. 

This may be counter-intuitive to most users and led to the issue during the development of another Spark  ML model: https://github.com/zhengruifeng/spark-libFM/issues/11. If a user actually does not require the training data set, it would be easier and more intuitive to use `RDD.sample`.
## How was this patch tested?

`build/mvn -DskipTests clean package` build succeeds

### CODE REVIEW

**Comment by srowen** [comment]
Jenkins test this please

File: mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala [hunk 1]
```diff
@@ -197,6 +197,11 @@ object GradientDescent extends Logging {
         "< 1.0 can be unstable because of the stochasticity in sampling.")
     }
 
+    if (numIterations * miniBatchFraction < 1.0) {
+      logWarning("Not all examples will be used if numIterations * miniBatchFraction " +
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala]: Use string interpolation to add the actual values into this warning message

Comment (Hydrotoast) [inline mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala]: Done

**Comment by srowen** [comment]
OK merged to master/2.0
